Id/test/Id.test.cpp

```diff
@@ -0,0 +1,154 @@
+#define CATCH_CONFIG_RUNNER
+
+// other includes
+#include "catch.hpp"
+
+// what we're testing
+#include "utility/Id.hpp"
+
+int testNumber = 0;
+
+namespace {
+
+  // test variable
+  std::string str("myId");
+  std::string validstr("validId");
+
+  // dummy test class and validation function
+  class Dummy {};
+}
+
+// override validateId function for the Dummy class
+namespace utility {
+
+  template <> bool validateId<Dummy>(const std::string& id) noexcept {
+
+    if (id != validstr) {
+
+      return false;
+    }
+    return true;
```

Comment:
Should this be:
```
if (id != validstr) {
    return false;
}
else {
    return true;
}
```

I'm not sure if one way is better than another. Please comment.